### PR TITLE
Cleanup TODOs for OTP 21+

### PR DESCRIPTION
**Pull request by hauleth**

### CODE REVIEW

File: lib/elixir/src/elixir_dispatch.erl [hunk 1]
```diff
@@ -303,7 +303,7 @@ format_error({deprecated, Mod, Fun, Arity, Message}) ->
 
 %% TODO: Do not rely on erlang:module_loaded/1 on Erlang/OTP 21+.
```
Comment (josevalim) [inline lib/elixir/src/elixir_dispatch.erl]: Please remove the TODO! :D

File: lib/elixir/src/elixir_erl_for.erl [hunk 2]
```diff
@@ -177,22 +177,11 @@ build_into(Ann, Clauses, Expr, Into, Uniq, S) ->
 
 %% TODO: Remove this check once we support Erlang/OTP 21+ exclusively.
```
Comment (josevalim) [inline lib/elixir/src/elixir_erl_for.erl]: Please remove the TODO. :D

Comment (hauleth) [reply lib/elixir/src/elixir_erl_for.erl]: I pushed the fix in the same time as you have been commenting ;)

File: lib/elixir/src/elixir.hrl [hunk 3]
```diff
@@ -44,12 +44,8 @@
...
--ifdef(OTP_RELEASE).
-  -if(?OTP_RELEASE >= 22).
-    -define(NO_SPAWN_COMPILER_PROCESS, no_spawn_compiler_process).
-  -elif(?OTP_RELEASE >= 21).
-    -define(NO_SPAWN_COMPILER_PROCESS, dialyzer, no_spawn_compiler_process).
-  -endif.
--else.
+-if(?OTP_RELEASE >= 22).
+  -define(NO_SPAWN_COMPILER_PROCESS, no_spawn_compiler_process).
+-elif(?OTP_RELEASE >= 21).
```
Comment (fertapric) [inline lib/elixir/src/elixir.hrl]: Should this be just `else`?

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**